Trim unexpected info from pack.mcmeta

dsch7705 opened the pull request:
Fixes #4331 

This applies the same fix implemented in 5ebd386797bf7c65da079959d32b89cd3783eda0 to DataPackUtils::processMCMeta.

dsch7705 committed:
Trim unexpected info from pack.mcmeta

Signed-off-by: Dylan Schooner <dschooner05@gmail.com>

TheKodeToad commented on launcher/minecraft/mod/tasks/LocalDataPackParseTask.cpp:
```diff
@@ -168,10 +168,20 @@ bool processZIP(DataPack* pack, ProcessingLevel level)
 // https://minecraft.wiki/w/Tutorials/Creating_a_resource_pack#Formatting_pack.mcmeta
 bool processMCMeta(DataPack* pack, QByteArray&& raw_data)
 {
+    QJsonParseError parse_error;
```
In case it's needed in more places probably best to create a helper in Json.h? (Json::parseUpToError or something, also no need to make it throw a JsonException as that seems like a bad idea anyway since it's easy to forget to catch)

dsch7705 replied:
Should I update McClient::parseResponse to use the helper or save for another PR?

TheKodeToad replied:
Best to update it in this PR *IMO*

dsch7705 committed:
Add Json::parseUntilMalformed helper

Signed-off-by: Dylan Schooner <dschooner05@gmail.com>

Ryex reviewed (changes requested):
id prefer if this was explicitly checking for `QJsonParseError::GarbageAtEnd`
instead of ignoring all json errors. the function could also potentially return the "garbage" by a QString out param

dsch7705 committed:
Use Json::parseUntilMalformed in McClient::parseResponse

Signed-off-by: Dylan Schooner <dschooner05@gmail.com>

dsch7705 commented:
> id prefer if this was explicitly checking for `QJsonParseError::GarbageAtEnd` instead of ignoring all json errors.

Excellent point.

dsch7705 committed:
Check specifically for GarbageAtEnd error

- Out parameter for garbage data\n- Rename parseUntilMalformed to parseUntilGarbage

Signed-off-by: Dylan Schooner <dschooner05@gmail.com>
